request.go

```diff
@@ -210,8 +227,13 @@ func (r *Request) open(h Handlers, pkt requestPacket) responsePacket {
 	var err error
 	switch {
 	case flags.Write, flags.Append, flags.Creat, flags.Trunc:
-		r.Method = "Put"
-		r.state.writerAt, err = h.FilePut.Filewrite(r)
+		if openFileWriter, ok := h.FilePut.(OpenFileWriter); ok && flags.Read {
```

Comment:
Seems like we could use an 'if flags.Read { … }` surrounding block to shortcircuit even attempting the type conversion if we know we won’t need it. But I don’t think it will end up being any clearer code, without an early return in the block that makes the `openFileWriter.OpenFile(…)` call.